tests/infer/mcmc/test_hmc.py

```diff
@@ -181,16 +180,16 @@ def model(data):
 
 def test_bernoulli_beta():
     def model(data):
-        alpha = pyro.param('alpha', variable([1.1, 1.1], requires_grad=True))
-        beta = pyro.param('beta', variable([1.1, 1.1], requires_grad=True))
+        alpha = pyro.param('alpha', torch.tensor([1.1, 1.1]))
```

Comment:
Hmm it seems a bit un-idiomatic to call `pyro.param` with `requires_grad=False`. Can we discuss to determine whether `with torch.no_grad()` would be a suitable alternative solution that could (1) preserve our existing speed, but (2) allow more idiomatic use of grad-requiring params?

Reply:
Actually, this was written in the spirit of other SVI models, but there really is no need to use the `pyro.param` primitive here. We could just use regular variables, and that might make things clearer since we are not interacting with the param store at all. Does that sound reasonable?

```
        alpha = torch.tensor([1.1, 1.1])
        beta = torch.tensor([1.1, 1.1])
```

Additionally, we are just generating samples from the latent sites and do not need `requires_grad` to be set to true on these parameter sites at all. So overall, using `pyro.param` just seems to be adding confusion here.

The only down-side (arguable) is that we can't just swap out a model in SVI, and use `HMC` to generate samples from it, but the semantics are sufficiently different anyhow. What do you think?

Reply:
Thanks for addressing this. Let's merge and then discuss how to make it easy to swap out SVI for HMC.